packages/core-components/src/layout/Sidebar/SidebarSubmenuItem.tsx

```diff
@@ -132,7 +131,10 @@ export const SidebarSubmenuItem = (props: SidebarSubmenuItemProps) => {
   if (dropdownItems !== undefined) {
     dropdownItems.some(item => {
       const resolvedPath = resolvePath(item.to);
-      isActive = locationPathname === resolvedPath.pathname;
+      isActive =
+        locationPathname === resolvedPath.pathname &&
+        locationSearch === toSearch;
```

Comment:
I think it's best if we make this matching less strict, or we're likely to break some other use-cases. I'm thinking that IF the sidebar item points to a URL that has any number of query parameters, then the item will be considered active if the current location has matching entries for all of those same parameters. The current location is however allowed to contain additional parameters.

Couple of examples:

```ts
function isMatch(to: string, currentLocation: string) { \* TODO *\ }

isMatch('/catalog?kind=component', '/catalog?kind=component') === true // exact match
isMatch('/catalog?x=foo', '/catalog?x=foo&y=bar') === true // target is a subset, extra parameters are fine
isMatch('/catalog?x=foo', '/catalog?y=bar') === false // current location does not have x=foo
isMatch('/catalog?x=foo&y=bar', '/catalog?y=bar&x=foo') === true // order doesn't matter
isMatch('/catalog', '/catalog?y=bar&x=foo') === true // extra parameters are fine
```

Reply:
Actually this can be easily done with the lodash isMatch() function. The issue is this gets more complicated when we have nested objects in the query. E.g., filtering in the catalog requires something like: 
`filters: {kind: 'component', type: ['service']`

Reply:
Would be worthwhile to make it this way anyways as due to the problem in the comment below, I don't think it will be possible to add items which do things like filtering the catalog yet